ext/drmgr/drmgr.h

```diff
@@ -452,6 +465,53 @@ drmgr_unregister_bb_instrumentation_ex_event(drmgr_app2app_ex_cb_t app2app_func,
                                              drmgr_insertion_cb_t insertion_func,
                                              drmgr_ilist_ex_cb_t instru2instru_func);
 
+DR_EXPORT
+/**
+ * Registers callback functions for the third
+ * instrumentation stage: instrumentation
+ * insertion.  drmgr will call \p func for each instruction with the
+ * specific opcode \p opcode.
+ *
+ * More than one call-back function can be mapped to the same opcode. Their
+ * execution sequence is determined by their priority \p priority if set. Ordering
+ * based on priority is also taken in to account with respect to insert bb events.
+ *
+ * Since this call-back is triggered during insert instrumentation,
+ * same usage rules apply. The call-back is allowed to insert meta
+ * instructions only immediately prior to the passed-in instruction.
+ * New non-meta instructions cannot be inserted.
+ *
+ * \return false upon failure.
+ *
+ * @param[in]  func  The opcode insertion callback to be called for the third
+ * stage for a specific opcode instruction. Cannot be NULL.
+ * @param[in]  opcode          The opcode to associate with the insertion callback.
+ * @param[in]  priority        Specifies the relative ordering of both callbacks.
+ *                             Can be NULL, in which case a default priority is used.
+ *
+ * \note It is possible that this call-back will be triggered for meta instructions.
+ * Therefore, we recommend that the call-back checks for meta instructions
```

Comment:
grammar: s/checks/check/

Reply:
I think it is correct how it is no?

Reply:
No, this is the subjunctive mood, which is not conjugated for subject.  Certainly it should agree with the other verb on the next line ("ignore").

Reply:
Ah, right, sorry. So the wording in the brackets needs to be fixed.

Reply:
Not sure what you mean about the brackets/parentheses: the "and ignore them, typically"?  I think that's fine.

Reply:
Haha, AFAIR in British English, it is acceptable to keep the s (but I need to change "ignore" to "ignores"). In AmE, this may come across as odd. So I'll go ahead and drop the "s" :p

Reply:
Done